pkg/validate/streaming.go

```diff
@@ -119,39 +147,52 @@ var _ = framework.KubeDescribe("Streaming", func() {
 	})
 })
 
-func createDefaultExec(c internalapi.RuntimeService, containerID string) string {
-	By("exec default command in container: " + containerID)
-	req := &runtimeapi.ExecRequest{
-		ContainerId: containerID,
-		Cmd:         []string{"echo", "hello"},
-		Stdout:      true,
-		Stderr:      true,
-	}
-
-	resp, err := c.Exec(req)
-	framework.ExpectNoError(err, "failed to exec in container %q", containerID)
+func createExec(c internalapi.RuntimeService, execReq *runtimeapi.ExecRequest) string {
+	By("exec given command in container: " + execReq.ContainerId)
+	resp, err := c.Exec(execReq)
+	framework.ExpectNoError(err, "failed to exec in container %q", execReq.ContainerId)
 	framework.Logf("Get exec url: " + resp.Url)
 	return resp.Url
 }
 
-func checkExec(c internalapi.RuntimeService, execServerURL string) {
+func checkExec(c internalapi.RuntimeService, execServerURL, stdout string, isTty bool) {
 	localOut := &bytes.Buffer{}
```

Comment:
Use safe buffer. https://github.com/umohnani8/cri-tools/blob/094e42ff5cde024a4e6bae544c1993e0f88d83a6/pkg/validate/streaming.go#L236

`bytes.Buffer` is not thread safe.

Actually use similar code with `checkAttach`. https://github.com/umohnani8/cri-tools/blob/094e42ff5cde024a4e6bae544c1993e0f88d83a6/pkg/validate/streaming.go#L257

Reply:
Good catch 👍